File: Baekjoon/Python/숫자 게임.py
import sys
input = sys.stdin.readline
from itertools import combinations

n = int(input())
ans = []
cards = [list(map(int,input().split())) for _ in range(n)]

for i in range(n):
    tmp = 0
    for j in combinations(cards[i],3):
        a = sum(j) % 10

        tmp  = max(tmp,a)
    
    ans.append((tmp,i+1))

result = sorted(ans,key=lambda x : (-x[0],-x[1]))
print(result[0][1])

File: Baekjoon/Java/숫자 게임.java
import java.io.*;
import java.util.*;


public class Main {
    private static StringTokenizer st;
    private static BufferedReader br;

    public static void main(String[] args) throws IOException{
        br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[][] arr = new int[n][5];
        int[] max = new int[n];

        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            for (int j = 0; j < 5; j++) {
                arr[i][j] = Integer.parseInt(st.nextToken());
            }
        }

        for (int i = 0; i < n; i++) {
            int sum = 0;
            int tmp = 0;
            for (int j = 0; j < 3; j++) {
                for (int k = j + 1; k < 4; k++) {
                    for (int l = k + 1; l < 5; l++) {
                        tmp = arr[i][j] + arr[i][k] + arr[i][l];
                        if (sum <= tmp % 10)
                            sum = tmp % 10;
                    }
                }
            }
            max[i] = sum;
        }

        int ans = 0;
        int result = max[0];

        for (int i = 0; i < n; i++) {
            if (max[i] >= result) {
                result = max[i];
                ans = i + 1;
            }
        }

        System.out.println(ans);
    }
}